src/librustc_lexer/src/lib.rs

```diff
@@ -236,12 +236,17 @@ pub enum Base {
 /// (e.g. "#![deny(missing_docs)]").
 pub fn strip_shebang(input: &str) -> Option<usize> {
     debug_assert!(!input.is_empty());
-    if !input.starts_with("#!") || input.starts_with("#![") {
+    let s: &str = &remove_whitespace(input);
+    if !s.starts_with("#!") || s.starts_with("#![") {
         return None;
     }
     Some(input.find('\n').unwrap_or(input.len()))
 }
 
+fn remove_whitespace(s: &str) -> String {
+    s.chars().filter(|c| !c.is_whitespace()).collect()
```

Comment:
This should use `is_whitespace` from `rustc_lexer` instead of the one from `libstd`.

Reply:
It also doesn't seem reasonable to filter and re-collect *all the program text* to check for something that almost never happens, or requires checking a couple of characters when it does.

Reply:
Looks like in this case
```rust
#!
[bad_attribute]
```
`#!` is no longer treated as a shebang anymore, which also seems incorrect.

Reply:
I think an empty shebang shouldn't be treated as a shebang, especially as it can be part of valid Rust syntax.

But I agree with everything else, and I wish this PR had been assigned to you...

Reply:
Just found #71487, will leave comments there.